File: SpaceGuildBack/c#/spaceship.cs
namespace SpaceGuild;
class Ship
{
    private int Health { get; set; } = 0;
    public Engine Engine { get; set; } = new Engine();
    public Weapon Weapon { get; set; } = new Weapon();
    public Shield Shield { get; set; } = new Shield();
    public Cargo Cargo { get; set; } = new Cargo();
    public Sensor Sensor { get; set; } = new Sensor();
    public StealthCloak StealthCloak { get; set; } = new StealthCloak();

    public Ship()
    {
        // Constructor remains empty
    }
}

File: SpaceGuildBack/spaceship.py
from spaceshipComponent import *

def Ship():
    return {
        'hp': 100,
        'engine': Engine('basic engine',0),
        'weapon': Weapon('basic gun',0),
        'shield': Shield('basic shield',0),
        'cargo': Cargo('basic cargo',0),
        'sensor': Sensor('basic sensor',0),
        'stealth_cloak': None
    }
